### PR TITLE
functional tests

### PR DESCRIPTION
**Pull request by Jennings-HMCTS**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -240,6 +243,15 @@ dependencies {
...
   }
 
+  functionalTestCompile sourceSets.test.runtimeClasspath
+  functionalTestAnnotationProcessor 'org.projectlombok:lombok:1.18.12'
+  functionalTestCompile group: 'net.serenity-bdd', name: 'serenity-core', version: versions.serenity
+  functionalTestCompile group: 'net.serenity-bdd', name: 'serenity-cucumber', version: versions.serenityCucumber
+  functionalTestCompile group: 'net.serenity-bdd', name: 'serenity-junit', version: versions.serenity
+  functionalTestCompile group: 'net.serenity-bdd', name: 'serenity-rest-assured', version: versions.serenity
+  functionalTestCompile group: 'net.serenity-bdd', name: 'serenity-spring', version: versions.serenity
+  compile group: 'org.jboss.aerogear', name: 'aerogear-otp-java', version:'1.0.0'
```
Comment (pats-john) [inline build.gradle:253]: What is this library(aerogear) used for???

File: build.gradle [hunk 2]
```diff
@@ -240,6 +243,15 @@ dependencies {
     exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
   }
 
+  functionalTestCompile sourceSets.test.runtimeClasspath
+  functionalTestAnnotationProcessor 'org.projectlombok:lombok:1.18.12'
```
Comment (pats-john) [inline build.gradle:247]: One more dependency would be required functionalTestCompileOnly 'org.projectlombok:lombok:1.18.12'

File: src/functionalTest/java/uk/gov/hmcts/reform/refunds/config/IdamService.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            .encoder(new JacksonEncoder())
+            .decoder(new JacksonDecoder())
+            .target(IdamApi.class, testConfig.getIdamApiUrl());
+    }
+
+
+    public ValidUser createUserWith(String userGroup, String... roles) {
+        String email = nextUserEmail();
+        CreateUserRequest userRequest = userRequest(email, userGroup, roles);
+        LOG.info("idamApi : " + idamApi.toString());
```
Comment (pats-john) [inline src/functionalTest/java/uk/gov/hmcts/reform/refunds/config/IdamService.java:45]: We should use the logging reccomended  by HMCTS.

File: src/functionalTest/java/uk/gov/hmcts/reform/refunds/config/S2sTokenService.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public S2sTokenService(OneTimePasswordFactory oneTimePasswordFactory, TestConfigProperties testProps) {
+        this.oneTimePasswordFactory = oneTimePasswordFactory;
+        s2sApi = Feign.builder()
+            .encoder(new JacksonEncoder())
+            .target(S2sApi.class, testProps.getS2sBaseUrl());
+    }
+
+    public String getS2sToken(String microservice, String secret) {
+        String otp = oneTimePasswordFactory.validOneTimePassword(secret);
+        LOG.info("s2sApi : " + s2sApi.toString());
```
Comment (pats-john) [inline src/functionalTest/java/uk/gov/hmcts/reform/refunds/config/S2sTokenService.java:27]: Logging comment as above.

**Review by pats-john — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (Jennings-HMCTS) [reply build.gradle:253]: It is used for generating OTP in ServiceAuthorization token.

File: build.gradle (hunk 2 above)
Comment (Jennings-HMCTS) [reply build.gradle:247]: dependency added

**Pull request merged**